src/gui/editorwidgets/core/qgseditorwidgetwrapper.cpp

```diff
@@ -49,6 +49,11 @@ QVariant QgsEditorWidgetWrapper::defaultValue() const
   return mDefaultValue;
 }
 
+void QgsEditorWidgetWrapper::resetValue()
+{
+  setValue( defaultValue() );
```

Comment:
Now, if a joined feature does not exist, the default value will be shown?

Imagining a joined feature with a "creation date" with default "now()"... I think in this case if the foreign key field is NULL, the creation_date should also be NULL (rather than today's date), no?